yodeploy/test_integration/helpers.py

```diff
@@ -58,15 +62,17 @@ def build_sample(app_name, version='1'):
         'PYTHONPATH': ':'.join(sys.path),
         'BUILD_NUMBER': version,
         'GIT_BRANCH': 'master',
-        'YOLA_SRC': os.path.join(tests_dir, 'samples')
+        'YOLA_SRC': os.path.join(tests_dir, 'samples'),
+        'PYTHONIOENCODING': 'utf-8'
```

Comment:
Sorry for taking so long to respond to the PYTHONIOENCODING things in particular. This was one of the most confusing issues for me when I was working this and I needed to backtrack quite a bit and recreate the scripts I originally used to reproduce this issue and explore solutions.

I added the `PYTHONIOENCODING` env data in 3 places, but this is the only one where it was actually fixing a real problem, IIRC. The other cases were either defensive, trying to make things consistent, or both. As you pointed out elsewhere, `build_artifact` isn't called from the other subprocesses. However, they could hypothetically run into similar issues if they attempted to output their own unicode characters. I think that means this boils down to two issues here:

1. Is the PYTHONIOENCODING thing necessary for `build_artifact` and situations similar to it or is there a better way of handling this?
2. If it is, do we want to defensively include it for our other yodeploy subprocess calls to mitigate the risk of running into similar issues?

I have a test scaffold setup on my machine to let us play around with solutions to issue 1. Do you have time to pair with me today? I'd love to talk through this with you and hopefully come up with something better than what I'm doing here.